src/Avalonia.Controls/ToggleSwitch.cs

```diff
@@ -14,6 +14,17 @@ static ToggleSwitch()
         {
             OffContentProperty.Changed.AddClassHandler<ToggleSwitch>((x, e) => x.OffContentChanged(e));
             OnContentProperty.Changed.AddClassHandler<ToggleSwitch>((x, e) => x.OnContentChanged(e));
+            IsCheckedProperty.Changed.AddClassHandler<ToggleSwitch>((x, e) =>
+            {
+                if ((e.NewValue != null) && (e.NewValue is bool val))
+                    x.UpdateKnobPos(val);
+            });
+
+            BoundsProperty.Changed.AddClassHandler<ToggleSwitch>((x, e) =>
```

Comment:
@grokys I think such handlers could be just done in layout arrange, right?

Reply:
@MarchingCube I tried that. No luck.

Reply:
you should be able to override ArrangeOverride... and at that point Bounds is valid... then run your code there.